src/libs/ReportUtils.ts

```diff
@@ -3465,13 +3471,54 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
 }
 
 /**
- * Takes several pieces of data from Onyx and evaluates if a report should be shown in the option list (either when searching
- * for reports or the reports shown in the LHN).
+ * Checks to see if a report's parentAction is a money request that contains a violation
+ */
+function doesTransactionThreadHaveViolations(report: Report, transactionViolations: TransactionViolations, reportActions: OnyxCollection<ReportActions>): boolean {
+    const resolvedParentReportAction = reportActions?.[`${report.parentReportID}`]?.[`${report.parentReportActionID}`];
+    if (!resolvedParentReportAction) {
+        return false;
+    }
+    if (resolvedParentReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
+        return false;
+    }
+    const {IOUTransactionID, IOUReportID} = resolvedParentReportAction?.originalMessage;
+    if (!IOUTransactionID || !IOUReportID) {
+        return false;
+    }
+    if (!isCurrentUserSubmitter(IOUReportID)) {
+        return false;
+    }
+    if (report?.stateNum !== CONST.REPORT.STATE_NUM.OPEN && report?.stateNum !== CONST.REPORT.STATE_NUM.PROCESSING) {
+        return false;
+    }
+    return hasViolation(IOUTransactionID, transactionViolations);
+}
+
+/**
+ * Checks to see if a report contains a violation
+ */
+function hasViolations(reportID: string, transactionViolations: TransactionViolations): boolean {
+    const transactions = TransactionUtils.getAllReportTransactions(reportID);
+    return transactions.some((transaction) => hasViolation(transaction.transactionID, transactionViolations));
+}
+
+/**
+ * Takes several pieces of data from Onyx and evaluates if a report should be shown in the option list (either when
+ * searching for reports or the reports shown in the LHN).
  *
- * This logic is very specific and the order of the logic is very important. It should fail quickly in most cases and also
- * filter out the majority of reports before filtering out very specific minority of reports.
+ * This logic is very specific and the order of the logic is very important. It should fail quickly in most cases and
+ * also filter out the majority of reports before filtering out very specific minority of reports.
  */
-function shouldReportBeInOptionList(report: OnyxEntry<Report>, currentReportId: string, isInGSDMode: boolean, betas: Beta[], policies: OnyxCollection<Policy>, excludeEmptyChats = false) {
+function shouldReportBeInOptionList(
+    report: OnyxEntry<Report>,
+    currentReportId: string,
+    isInGSDMode: boolean,
+    betas: Beta[],
+    policies: OnyxCollection<Policy>,
+    allReportActions?: OnyxCollection<ReportActions>,
+    excludeEmptyChats = false,
+    transactionViolations?: TransactionViolations,
```

Comment:
- Can we make allReportAction and transactionViolations non-optional?
- Any reason why you added allReportActions _before_ excludeEmptyChats and transactionViolations _after_ excludeEmptyChats?

This isn't a blocker, I'm just curious

Reply:
no idea -- it's an easy refactor though.

Reply:
done.